toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordChallenge.kt

```diff
@@ -39,6 +40,8 @@ public class UsernamePasswordChallenge(
     private var onCancel: (() -> Unit)? = onCancel
     private val _additionalMessage: MutableStateFlow<String?> = MutableStateFlow(null)
     public val additionalMessage: StateFlow<String?> = _additionalMessage.asStateFlow()
+    public val hostname: String? = Uri.parse(url).host
+
 
```

Comment:
We don't want this to be nullable, we can guard it as such: 
```suggestion
    public val hostname: String by lazy {
        Uri.parse(url).host ?: url
    }
```